presto-main/src/main/java/com/facebook/presto/operator/PagesIndex.java

```diff
@@ -469,4 +471,23 @@ public String toString()
                 .add("estimatedSize", estimatedSize)
                 .toString();
     }
+
+    public Iterator<Page> getPages(int[] outputChannels)
+    {
+        return new AbstractIterator<Page>() {
+            private int currentPosition = 0;
+            private PageBuilder pageBuilder = new PageBuilder(types);
+
+            @Override
+            public Page computeNext()
+            {
+                pageBuilder.reset();
```

Comment:
I think you should call `reset` after `pageBuilder.build()`. See for instance com.facebook.presto.operator.WindowOperator#extractOutput